aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,626 @@
+spot: Spot's base frame (defined by current location of spot). This frame is published by Proprioception node in local_launch shell script
+marker: Marker frame (defined by the center of spot dock's april tag). This frame is published by SpotAriaP1 node at the start of the node when spot looks at the marker and detects it
+ariaWorld: Aria's world frame (defined by location where Aria's SLAM pipeline runs). This frame is published by AriaLiveReader node
+aria: Aria's cpf frame (defined by location of Aria's camera). This frame is published by AriaLiveReader node everytime it receives a pose from Aria.
+
+Extra Aria frames (for information only):
+cpf: Center frame of Aria with Z pointing out, X pointing up and Y pointing left
+device: Base frame of Aria which is aligned with left-slam camera frame
+"""
+
+
+class AriaLiveReader:
+    """
+    Class to livestream frames and publish different data on ROS topics
+    """
+
+    def __init__(self, verbose: bool = False) -> None:
+        super().__init__()
+        self.verbose = verbose
+        self._in_index = 0
+        self._out_index = 0
+        self._is_connected = False
+        self.aria_open_loop_pose_sub = rospy.Subscriber(
+            "/pose_dynamics", PoseStamped, self.on_pose_received
+        )
+        self.pose_of_interest_publisher = rospy.Publisher(
+            "/aria_pose_of_interest", PoseStamped, queue_size=10
+        )
+        self.aria_odom_publisher = rospy.Publisher(
+            "/aria_odom", Odometry, queue_size=10
+        )
+        self.aria_cpf_publisher = rospy.Publisher(
+            "/aria_current_pose", PoseStamped, queue_size=10
+        )
+        self.pose_of_interest_marker_publisher = rospy.Publisher(
+            "/aria_pose_of_interest_marker", Marker, queue_size=10
+        )
+
+        self.aria_pose = None
+        self.aria_ros_pose = None
+        self.aria_rgb_frame = None
+        aria.set_log_level(aria.Level.Info)
+
+        # 1. Create StreamingClient instance
+        self._sensors_calib_json = None
+        self._sensors_calib: Optional[aria_core.calibration.SensorCalibration] = None
+        self._device_T_cpf = None
+        self._rgb_calib_params: Optional[aria_core.calibration.CameraCalibration] = None
+        self._dst_calib_params: Optional[aria_core.calibration.CameraCalibration] = None
+        self.device_T_camera: Optional[sp.SE3] = None
+        self.device, self.device_client = self._setup_aria()
+
+        self.static_tf_broadcaster = StaticTransformBroadcaster()
+
+        # Maintain a list of all poses where qr code is detected (w.r.t ariaWorld)
+        self.marker_positions_list = (
+            []
+        )  # type: List[np.ndarray] # List of  position as np.ndarray (x, y, z)
+        self.marker_quaternion_list = (
+            []
+        )  # type: List[np.ndarray] # List of quaternions as np.ndarray (x, y, z, w)
+        self.fix_list_length = 10  # TODO: Add this to config
```

Comment:
please fix this TODO. Thank you.

Reply:
updated. currently hard set the value inside math_utils.py